snippets/crypto-derive.ts

```diff
@@ -96,9 +101,44 @@ tabris.onLog(({message}) => stack.append(TextView({text: message})));
       size * 8
     );
   }
+}
+
+async function generateDeriveEncryptAndDecrypt({inTee, usageRequiresAuth} = {inTee: false, usageRequiresAuth: false}) {
+  const ecdhP256 = {name: 'ECDH' as const, namedCurve: 'P-256' as const};
+  const aesGcm = {name: 'AES-GCM' as const};
+
+  // Generate Alice's ECDH key pair
+  const alicesKeyPair = await crypto.subtle.generateKey(ecdhP256, true, ['deriveBits']);
+
+  // Generate Bob's ECDH key pair
+  const bobsKeyPair = await crypto.subtle.generateKey(ecdhP256, true, ['deriveBits'], {inTee, usageRequiresAuth});
+
+  // Derive Alice's AES key
+  const alicesAesKey = await deriveAesKey(bobsKeyPair.publicKey, alicesKeyPair.privateKey, 'encrypt');
+
+  // Derive Bob's AES key
+  const bobsAesKey = await deriveAesKey(alicesKeyPair.publicKey, bobsKeyPair.privateKey, 'decrypt');
 
-  function test(name, actual, expected) {
-    console.log(name, expected === actual ? 'OK' : `expected: ${expected}, actual: ${actual}`);
+  // Encrypt a message with Alice's AES key
+  const message = await new Blob(['Message']).arrayBuffer();
+  const iv = crypto.getRandomValues(new Uint8Array(12));
+  const ciphertext = await crypto.subtle.encrypt({...aesGcm, iv}, alicesAesKey, message);
```

Comment:
Maybe rename `ciphertext` to `cipherText`?

Reply:
Seems to be an established term, so I would leave it as-is: https://en.wikipedia.org/wiki/Ciphertext